clinica/pipelines/pet_volume/pet_volume_pipeline.py

```diff
@@ -20,111 +21,11 @@
 
 
 class PETVolume(cpe.Pipeline):
-    """PETVolume - Pre-processing of PET images using SPM.
-
-    Args:
-        input_dir: A BIDS directory.
-        output_dir: An empty output directory where CAPS structured data will be written.
-        subjects_sessions_list: The Subjects-Sessions list file (in .tsv format).
+    """PETVolume - Volume-based processing of PET images using SPM.
 
     Returns:
         A clinica pipeline object containing the PETVolume pipeline.
-
-    Raises:
-
-
     """
-
-    def __init__(self,
-                 bids_directory=None,
-                 caps_directory=None,
-                 tsv_file=None,
-                 base_dir=None,
-                 name=None,
-                 group_id='default',
-                 fwhm_tsv=None):
-        from pandas.io.parsers import read_csv
-        import os
-        from clinica.utils.exceptions import ClinicaCAPSError
-        from colorama import Fore
-
-        super(PETVolume, self).__init__(
-            bids_directory=bids_directory,
-            caps_directory=caps_directory,
-            tsv_file=tsv_file,
-            base_dir=base_dir,
-            name=name)
-
-        if not group_id.isalnum():
-            raise ClinicaCAPSError(Fore.RED + '[Error] Not valid group_id value. It must be composed only by letters and/or numbers' + Fore.RESET)
-
-        if not os.path.isdir(os.path.join(os.path.abspath(caps_directory), 'groups')):
-            raise ClinicaCAPSError(Fore.RED + '[Error] There is no \'groups\' folder in your CAPS directory. (Have you run t1-volume pipeline ?)' + Fore.RESET)
-
-        # Check that group already exists
-        if not os.path.exists(os.path.join(os.path.abspath(caps_directory), 'groups', 'group-' + group_id)):
-            error_message = group_id \
-                            + ' group does not exists, please choose an other one (or maybe you need to run t1-volume-create-dartel).' \
-                            + '\nGroups that already exist in your CAPS directory are: \n'
-            list_groups = os.listdir(os.path.join(os.path.abspath(caps_directory), 'groups'))
-            is_empty = True
-            for e in list_groups:
-                if e.startswith('group-'):
-                    error_message += '\t' + e + ' \n'
-                    is_empty = False
-            if is_empty is True:
-                error_message += 'NO GROUP FOUND'
-            raise ClinicaCAPSError(Fore.RED + '[Error] ' + error_message + Fore.RESET)
```

Comment:
These lines have disappeared. I think this check is important

Reply:
I forgot to move:
```python
        if not group_id.isalnum():
            raise ClinicaCAPSError(Fore.RED + '[Error] Not valid group_id value. It must be composed only by letters and/or numbers' + Fore.RESET)

        if not os.path.isdir(os.path.join(os.path.abspath(caps_directory), 'groups')):
            raise ClinicaCAPSError(Fore.RED + '[Error] There is no \'groups\' folder in your CAPS directory. (Have you run t1-volume pipeline ?)' + Fore.RESET)
```
to adequate locations indeed.



I have added new commits `T1Volume*` / `PETVolume`  pipelines:
1)
```python
        if not group_id.isalnum():
            raise ClinicaCAPSError(Fore.RED + '[Error] Not valid group_id value. It must be composed only by letters and/or numbers' + Fore.RESET)
```
is now in `check_pipeline_parameters()` methods and replaced by `check_group_label()` function.
2)
```python
        if not os.path.isdir(os.path.join(os.path.abspath(caps_directory), 'groups')):
            raise ClinicaCAPSError(Fore.RED + '[Error] There is no \'groups\' folder in your CAPS directory. (Have you run t1-volume pipeline ?)' + Fore.RESET)

        # Check that group already exists
        if not os.path.exists(os.path.join(os.path.abspath(caps_directory), 'groups', 'group-' + group_id)):
            error_message = group_id \
                            + ' group does not exists, please choose an other one (or maybe you need to run t1-volume-create-dartel).' \
                            + '\nGroups that already exist in your CAPS directory are: \n'
            list_groups = os.listdir(os.path.join(os.path.abspath(caps_directory), 'groups'))
            is_empty = True
            for e in list_groups:
                if e.startswith('group-'):
                    error_message += '\t' + e + ' \n'
                    is_empty = False
            if is_empty is True:
                error_message += 'NO GROUP FOUND'
            raise ClinicaCAPSError(Fore.RED + '[Error] ' + error_message + Fore.RESET)
```
was replaced by:
```python
        # Check that group already exists
        if not exists(join(self.caps_directory, 'groups', 'group-' + self.parameters['group_id'])):
            print_groups_in_caps_directory(self.caps_directory)
            raise ClinicaException(
                '%sGroup %s does not exist. Did you run t1-volume or t1-volume-create-dartel pipeline?%s' %
                (Fore.RED, self.parameters['group_id'], Fore.RESET)
            )
```